### DISCUSSION TITLE
Link to existing notes?

### DISCUSSION THREAD

**Discussion by p-wieland** [zk-org/zk-nvim#74] [Help] [answered]
Hi,

I really like the plugin, helps me a lot! Thanks for all your work! 
Maybe I missed it but is it possible to create links to existing notes? I imagine something like this:
- select text which is used as link description
- hit key
- a picker pops up and let's you choose the note which should be linked

Is this somehow possible?

**Comment by mickael-menu** [comment]
It's not yet possible although I think it would be a useful feature to have.

However you should be able to link to existing notes using the auto-completion by hitting `[[`.

**Reply by tevaughan** [reply]
I am new to this and probably just confused, but doesn't ZkInsertLinkAtSelection do exactly what p-wieland wants?

I have `<leader>zil` mapped to `<Cmd>ZkInsertLinkAtSelection<CR>` in 'after/ftplugin/markdown.lua', and it seems mostly to work.

I was thinking about opening a new discussion to address my issue, but perhaps I'll just try asking it here because it's related:

How do I get the target (what's inside the parentheses, as I'm using markdown-style links) to include not just the path to the file-stem but also to have the '.md' extension at the end?

As it is everything works well enough in neovim, but I'd like my markdown-files to work also in the browser on gitlab or bitbucket where I push them.

The link doesn't work in the browser if I don't manually add the '.md' to each target when I'm working in neovim.

Is there an easy way to get the full filename automatically inserted by ZkInsertLinkAtSelection?

**Reply by mickael-menu** [reply]
Yes, this new command was not available by the time @p-wieland asked their question.

> How do I get the target (what's inside the parentheses, as I'm using markdown-style links) to include not just the path to the file-stem but also to have the '.md' extension at the end?

With the [`link-drop-extension`](https://github.com/mickael-menu/zk/blob/main/docs/note-format.md#markdown) config property.

```toml
[format.markdown]
link-drop-extension = false
```

**Reply by tevaughan** [reply]
Thanks!
(I should have gone more carefully through `config.toml` before asking about this.)

**Reply by mickael-menu** [reply]
No problem, I'm counting on some AI tool reading the manuals for us soon 😄

**Comment by mickael-menu** [comment, marked as answer]
This feature is now available with the command `ZkInsertLinkAtSelection`.
